### PR TITLE
feat: solana connector

### PR DESCRIPTION
**Pull request by LeoCourbassier**
* Adds Solana Connector using Web3Modal/solana package.
* Adds basic tests to Solana Connector.

### CODE REVIEW

File: packages/solana-connector/src/utils/solanaProvider.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import {
+  type Web3Modal,
+  createWeb3Modal,
+  defaultSolanaConfig,
+} from '@web3modal/solana';
+import { SolanaChains } from '../constants';
+import type { SolanaConfig } from '../types';
+
+const DEFAULT_PROJECT_ID = 'a424aaaa79c1949c6bf08f3c0874bed3';
```
Comment (luizstacio) [inline packages/solana-connector/src/utils/solanaProvider.ts:9]: As a place holder this should be empty or fill with `0`s

**Review by luizstacio — CHANGES_REQUESTED** [review]

File: packages/solana-connector/src/utils/Predicate.ts [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+      blockCreated: bn(),
+      txCreatedIdx: bn(),
+    });
+    fakePredicate.populateTransactionPredicateData(request);
+    const txId = request.getTransactionId(chainId);
+    const signature = nacl.sign.detached(
+      this.getSmallTxId(txId),
+      keypair.secretKey,
+    );
+    request.witnesses = [signature, keypair.publicKey.toBytes()];
```
Comment (luizstacio) [inline packages/solana-connector/src/utils/Predicate.ts:107]:
```suggestion
    request.witnesses = [signature];
```

Comment (luizstacio) [inline packages/solana-connector/src/utils/Predicate.ts:107]: We could move this part to be a separate function on the class `populateFakeSignature` in this way once you starting refactoring other predicates you can move this to a common package and extend the class overriding only the `populateFakeSignature`.

File: packages/solana-connector/src/utils/Predicate.ts [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  private bytecode: Uint8Array;
+
+  constructor({ abi, bytecode }: PredicateConfig) {
+    this.abi = abi;
+    this.bytecode = bytecode;
+  }
+
+  getPredicateAddress = memoize((svmAddress: string): string => {
+    const configurable = {
+      SIGNER: hexlify(bs58.decode(svmAddress)),
```
Comment (luizstacio) [inline packages/solana-connector/src/utils/Predicate.ts:33]: We could move this part to be a separate function on the class `getAddress()`. In this way, once you start refactoring other predicates, you can move this to a common package and extend the class overriding only the `getAddress`.

**Review by luizstacio — CHANGES_REQUESTED** [review]

File: packages/solana-connector/src/utils/Predicate.ts (hunk 2 above)
Comment (LeoCourbassier) [reply packages/solana-connector/src/utils/Predicate.ts:107]: I have already done this in the refactor PR. I'm just waiting for this PR to be merged to open the other PR.

File: packages/solana-connector/src/utils/Predicate.ts (hunk 3 above)
Comment (LeoCourbassier) [reply packages/solana-connector/src/utils/Predicate.ts:33]: Same as the comment above

File: packages/solana-connector/src/utils/Predicate.ts (hunk 2 above)
Comment (LuizAsFight) [reply packages/solana-connector/src/utils/Predicate.ts]: @luizstacio this can be done on the refactor PR

**Review by LuizAsFight — CHANGES_REQUESTED** [review]

**Review by LuizAsFight — CHANGES_REQUESTED** [review]
functional:

on the PR link, and also when running locally, when I click on "Connect Wallet" -> "Solana Wallets" , nothing happens

File: packages/connectors/src/solana-connector/index.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+export * from '@fuel-connectors/solana-connector';
```
[the remark this answers is not in the document]
Comment (LeoCourbassier) [reply packages/connectors/src/solana-connector/index.ts:1]: I tried removing but it was actually needed

**Comment by LeoCourbassier** [comment]
image
image

**Review by LuizAsFight — CHANGES_REQUESTED** [review]
functional:

1- after refreshing the page:
1.1- user lose connection
1.2- when trying to connect again, I see this popup

image

**Review by LuizAsFight — APPROVED** [review]

**Review by luizstacio — APPROVED** [review]

**Pull request merged**